flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/AggregationCodeGenerator.scala

```diff
@@ -161,13 +182,108 @@ class AggregationCodeGenerator(
         }
     }
 
+    /**
+      * Adds a reusable [[org.apache.flink.table.api.dataview.DataView]] to the open, cleanup,
+      * close and member area of the generated function.
+      *
+      */
+    def addReusableDataViews: Unit = {
+      if (accConfig != null && accConfig.isStateBackedDataViews) {
+        val descMapping: Map[String, StateDescriptor[_, _]] = accConfig.accSpecs
+          .flatMap(specs => specs.map(s => (s.id, s.toStateDescriptor)))
+          .toMap[String, StateDescriptor[_, _]]
+
+        for (i <- aggs.indices) yield {
+          for (spec <- accConfig.accSpecs(i)) yield {
+            val dataViewField = spec.field
+            val dataViewTypeTerm = dataViewField.getType.getCanonicalName
+            val desc = descMapping.getOrElse(spec.id,
+              throw new CodeGenException(s"Can not find ListView in accumulator by id: ${spec.id}"))
+
+            // define the DataView variables
+            val serializedData = AggregateUtil.serialize(desc)
```

Comment:
Why do we need to serialize and deserialize state descriptors? Can't we just generate the code to instantiate them? IMO, that would be more straightforward and easier to debug.

Reply:
state descriptors need TypeInformation which passed by user,  so we can not code gen state descriptor.